libraries/state_history/include/eosio/state_history/log.hpp

```diff
@@ -59,14 +59,18 @@ struct state_history_log_header {
 static const int state_history_log_header_serial_size = sizeof(state_history_log_header::magic) +
                                                         sizeof(state_history_log_header::block_id) +
                                                         sizeof(state_history_log_header::payload_size);
+struct state_history_log_prune_config {
+   uint32_t                prune_blocks;                  //number of blocks to prune to when doing a prune
+   size_t                  prune_threshold = 4*1024*1024; //(approximately) how many bytes need to be added before a prune is performed
+   std::optional<size_t>   vacuum_on_close;               //when set, a vacuum is performed on dtor if log contains less than this many bytes
```

Comment:
Did you not want to keep the default behavior of vacuuming on exit assuming the remaining size is less than 1 GiB? That behavior has now changed with these latest commits since this config is not set accordingly in state_history_plugin.cpp either (only in the tests).

Reply:
I should have maintained that just for consistency on the update. But there appears to still be some indecision on what we should do wrt automatic vacuuming. (I thought I was well opinionated on the matter before so many other opinions got mixed in). Let's decide for certain on 13th

Reply:
Okay,so it seems like we decided to not automatically vacuum on exit.

Reply:
Right, we've decided not to because the size of the logs are so big that even at 100000 blocks (less than a day) you'll be waiting minutes to vacuum on consumer tier hardware.